src/transform.js

```diff
@@ -41,12 +43,13 @@ function _transform (node, transformations, o = {}, property, parent) {
 		}
 		node = transformedNode;
 
-		parents.set(node, parent, {force: true});
 		const properties = childProperties[node.type] ?? [];
 		for (const prop of properties) {
 			node[prop] = _transform(node[prop], transformations, o, prop, node);
 		}
 	}
 
+	parents.clear(node);
```

Comment:
I called this in here (`_transform()`) instead of using `parents.clearAll` in `transform()` for performance reasons, but we can switch to the latter if you'd rather

Reply:
My main worry would be that this means everything that runs within the callback will have incorrect parent pointers until the very end.